tests/test_jit.py

```diff
@@ -28,16 +28,18 @@ def test_filter_jit(donate, getkey):
         eqx.nn.MLP(2, 2, 2, 2, key=getkey()),
     ]
     array_tree = [{"a": a, "b": b}, (c,)]
-    mlp_add = jtu.tree_map(lambda u: u + 1 if eqx.is_array(u) else u, general_tree[-1])
+    mlp_add = jtu.tree_map(
+        lambda u: jnp.asarray(u) + 1 if eqx.is_array(u) else u, general_tree[-1]
```

Comment:
`is_array` checks for `np.generic`, but this doen't support `u+1`. Do you want to restrict from `np.generic` to `np.number | np.bool | ...` ?